lib/msal-angular/src/msal.interceptor.ts

```diff
@@ -3,61 +3,82 @@
  * Licensed under the MIT License.
  */
 
-import { Injectable } from "@angular/core";
 import {
     HttpRequest,
     HttpHandler,
     HttpEvent,
-    HttpInterceptor,
-    HttpErrorResponse
+    HttpInterceptor
 } from "@angular/common/http";
-
-import { Observable, from } from "rxjs";
-import { mergeMap, tap } from "rxjs/operators";
-
-import {MsalService} from "./msal.service";
-import { BroadcastService } from "./broadcast.service";
-import { AuthResponse, ServerHashParamKeys } from "msal";
+import { Observable, EMPTY } from "rxjs";
+import { switchMap, catchError } from "rxjs/operators";
+import { MsalService } from "./msal.service";
+import { AuthenticationResult, BrowserConfigurationAuthError, InteractionType, StringUtils } from "@azure/msal-browser";
+import { Injectable, Inject } from "@angular/core";
+import { MSAL_INTERCEPTOR_CONFIG } from "./constants";
+import { MsalInterceptorConfiguration } from "./msal.interceptor.config";
 
 @Injectable()
 export class MsalInterceptor implements HttpInterceptor {
-    constructor(private auth: MsalService ,  private broadcastService: BroadcastService) {}
+    constructor(
+        @Inject(MSAL_INTERCEPTOR_CONFIG) private msalInterceptorConfig: MsalInterceptorConfiguration,
+        private authService: MsalService
+    ) {}
 
     intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
```

Comment:
We should address these linter warnings at some point after this PR

Reply:
Yes, definitely!